proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,315 @@
+request](https://github.com/aws/aws-cli/pull/5425). No further interaction from
+the maintainers has occurred.
+
+The current review process is very *ad hoc*. The AWS CLI team has limited
+bandwidth and competing priorities; without well-defined next steps for the
+maintainers, these types of requests are then the first to be neglected for
+prioritization. A community contribution should not be left in an incomplete
+state or waiting on a maintainer; it should be resolved by merging it or closing
+it. When an issue is raised and reviewed, we need to prioritize it and see it
+through.
+
+#### Competing priorities and limited bandwidth
+
+There are 151 open issues that have upvotes or reactions from five or more
+users, indicating broader interest. That volume is not tractable for the
+maintainers to implement themselves. Community members need to be able to
+contribute more easily while balancing the efforts of maintainers to conduct
+thorough reviews of code before merging.
+
+We need to collect and review user feedback to decide on what to work on to
+determine what contributions are of interest. We need to tell users how they can
+provide feedback in the best way for us to take action. We also need to
+strategically and fairly manage the amount of in flight work so that open
+requests do not pile up.
+
+#### Pull requests with limited background and context
+
+Many pull requests come as unsolicited changes that often do not correspond to
+an open issue that has been discussed by the maintainers or the community. They
+often end up lingering with no communication from the maintainers to indicate
+the feasibility or appropriateness of the changes. As an example, [`aws-cli`
+pull request 2636](https://github.com/aws/aws-cli/pull/2636)  adds functionality
+to configure MFA. It adds or alters 231 lines of code across five files, and is
+a significant new feature. There is no open issue or discussion about the
+feature other than on the PR. It has no comments from AWS employees, other than
+to indicate that it is a “large” pull request, since 2017. Conversation
+continued for a year on the PR by interested users with no other interaction or
+guidance from AWS.
+
+A idea that has not been vetted by the community and the maintainers is
+difficult to approve; it's hard to have a conversation when it involves an idea
+proposal as well as code changes. It usually ends in an unmerged pull request or
+significant code revision, which is effort that was unnecessarily expended by
+the user and potentially the maintainer. We need a structured process for the
+life cycle of an idea that can culminate in a community contribution. We also
+need to prioritize issues against other requests from the community given the
+effort to contribute for users and review for maintainers.
+
+## Specification
+
+This section defines:
+
+* [Current process](#current-process)
+* [Proposed improvements](#proposed-improvements)
+
+### Current Process
```

Comment:
For the current process, it probably makes sense to put it in the background/motivation section. For the Specification section, the general rule of thumb is to only outline the specific technical changes that will be happening as any background information should have been covered in the previous sections.

Reply:
Done!